Model/Product/Index/Index.php

```diff
@@ -0,0 +1,245 @@
+    const DB_VALUE_BOOLEAN_FALSE = "0";
+
+    /**
+     * @inheritdoc
+     */
+    public function getId()
+    {
+        return $this->getData(self::ID);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getProductId()
+    {
+        return $this->getData(self::PRODUCT_ID);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getInSync()
+    {
+        return $this->getData(self::IN_SYNC);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getIsDirty()
+    {
+        return $this->getData(self::IS_DIRTY);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getStoreId()
+    {
+        return $this->getData(self::STORE_ID);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getProductData()
+    {
+        return $this->getData(self::PRODUCT_DATA);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getCreatedAt()
+    {
+        return $this->getData(self::CREATED_AT);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getUpdatedAt()
+    {
+        return $this->getData(self::UPDATED_AT);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setId($id)
+    {
+        return $this->setData(self::ID, $id);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setProductId($productId)
+    {
+        return $this->setData(self::PRODUCT_ID, $productId);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setCreatedAt(\DateTime $createdAt)
+    {
+        return $this->setData(self::CREATED_AT, $createdAt);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function setUpdatedAt(\DateTime $updatedAt)
+    {
+        return $this->setData(self::UPDATED_AT, $updatedAt);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setInSync($inSync)
+    {
+        return $this->setData(self::IN_SYNC, $inSync ? self::DB_VALUE_BOOLEAN_TRUE : self::DB_VALUE_BOOLEAN_FALSE);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setIsDirty($isDirty)
+    {
+        return $this->setData(self::IS_DIRTY, $isDirty ? self::DB_VALUE_BOOLEAN_TRUE : self::DB_VALUE_BOOLEAN_FALSE);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setStoreId($storeId)
+    {
+        return $this->setData(self::STORE_ID, $storeId);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setProductData($productData)
+    {
+        return $this->setData(self::PRODUCT_DATA, $productData);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function setNostoProduct(NostoProductInterface $product)
+    {
+        return $this->setProductData(serialize($product));
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getNostoProduct()
+    {
+        try {
+            $unserialized = unserialize($this->getProductData(), [Product::class]); // @codingStandardsIgnoreLine
+        } catch (\Exception $exception) {
```

Comment:
Do we remove the exception? I don't see how this could happen except due to a broken upgrade. If it does break, I would rather have this error bubble to the surface than silently swallowing it.